code/modules/mob/living/carbon/brain/posibrain.dm

```diff
@@ -81,3 +82,36 @@
 	..()
 	src.brainmob.name = "[pick(list("PBU","HIU","SINA","ARMA","OSI"))]-[rand(100, 999)]"
 	src.brainmob.real_name = src.brainmob.name
+
+
+/obj/item/device/mmi/digital/posibrain/proc/healthcheck()
+	if(damageamount <=0)
+		playsound(loc, 'sound/items/countdown.ogg', 125, 1)
+		sleep(20)
+		playsound(loc, 'sound/effects/alert.ogg', 125, 1)
+		sleep(10)
+		new /obj/effect/gibspawner/robot(src.loc)
+		qdel(src)
+	return
+
+/obj/item/device/mmi/digital/posibrain/bullet_act(var/obj/item/projectile/Proj)
+
+	if(!Proj.nodamage)
+		switch(Proj.damage_type)
+			if(BRUTE)
+				healthcheck()
+				damageamount -= (rand(5,10))
+			if(BURN)
+				healthcheck()
+				damageamount -= (rand(25,30))
```

Comment:
20-30 damage reduction would most likely make laser rifle almost ineffective. I would make it 10-15 max.